File: test/unit/code_parser/whole_tests/views/dc_2014.cs
// Copyright (c) 2000 ActivePDF, Inc.
// ActivePDF DocConverter 2014
// Example generated 01/01/2000 

using System;

// Make sure to add the ActivePDF product .NET DLL(s) to your application.
// .NET DLL(s) are typically found in the products 'bin' folder.

class Examples
{
  public static void Example()
  {
    string strPath;
    DCDK.Results.DocConverterResult results;
    
    strPath = System.AppDomain.CurrentDomain.BaseDirectory;

    // Instantiate Object
    APDocConverter.DocConverter oDC = new APDocConverter.DocConverter();
    
    results = oDC.ConvertToPDF(strPath + "word.doc", strPath + "word.pdf");
    if (results.DocConverterStatus != DCDK.Results.DocConverterStatus.Success)
    {
      ErrorHandler("ConvertToPDF", results, results.DocConverterStatus.ToString());
    }
    
    results = oDC.ConvertFromPDF(strPath + "word.pdf", strPath + "word.pdf", oDC.FromPDFFormat = APDocConverter.FromPDFFunction.ToWord);
    if (results.DocConverterStatus != DCDK.Results.DocConverterStatus.Success)
    {
      ErrorHandler("ConvertFromPDF", results, results.DocConverterStatus.ToString());
    }
    
    // Release Object
    oDC = null;
    
    // Process Complete
    WriteResults("Done!");
  }
  
  // Error Handling
  public static void ErrorHandler(string strMethod, ADK.Results.Result results, string errorStatus)
  {
    WriteResults("Error with " + strMethod);
    WriteResults(errorStatus);
    WriteResults(results.Details);
    if (results.Origin.Function != strMethod)
    {
      WriteResults(results.Origin.Class + "." + results.Origin.Function);
    }
    if (results.ResultException != null)
    {
      // To view the stack trace on an exception uncomment the line below
      //WriteResults(results.ResultException.StackTrace);
    }
    Environment.Exit(1);
  }
  
  // Write output data
  public static void WriteResults(string content)
  {
    // Choose where to write out results
  
    // Debug output
    //System.Diagnostics.Debug.WriteLine("ActivePDF: * " + content);
  
    // Console
    Console.WriteLine(content);
  
    // Log file
    //using (System.IO.TextWriter writer = new System.IO.StreamWriter(System.AppDomain.CurrentDomain.BaseDirectory + "application.log", true))
    //{
    //    writer.WriteLine("[" + DateTime.Now.ToString("MM/dd/yyyy HH:mm:ss") + "]: => " + content);
    //}
  }
}

File: test/unit/code_parser/whole_tests/views/dc_2014.vb
' Copyright (c) 2000 ActivePDF, Inc.
' ActivePDF DocConverter 2014
' Example generated 01/01/2000 

Imports System

' Make sure to add the ActivePDF product .NET DLL(s) to your application.
' .NET DLL(s) are typically found in the products 'bin' folder.

Public Class Examples
  Sub Example()
    Dim strPath As String, results As DCDK.Results.DocConverterResult

    strPath = AppDomain.CurrentDomain.BaseDirectory

    ' Instantiate Object
    Dim oDC As APDocConverter.DocConverter = New APDocConverter.DocConverter()
    
    results = oDC.ConvertToPDF(strPath & "word.doc", strPath & "word.pdf")
    If results.DocConverterStatus <> DCDK.Results.DocConverterStatus.Success Then
      ErrorHandler("ConvertToPDF", results, results.DocConverterStatus.ToString())
    End If
    
    results = oDC.ConvertFromPDF(strPath & "word.pdf", strPath & "word.pdf", oDC.FromPDFFormat = APDocConverter.FromPDFFunction.ToWord)
    If results.DocConverterStatus <> DCDK.Results.DocConverterStatus.Success Then
      ErrorHandler("ConvertFromPDF", results, results.DocConverterStatus.ToString())
    End If
    
    ' Release Object
    oDC = Nothing
    
    ' Process Complete
    WriteResults("Done!")
  End Sub
  
  ' Error Handling
  Sub ErrorHandler(ByVal strMethod As String, ByVal results As ADK.Results.Result, ByVal errorStatus As String)
    WriteResults("Error with " + strMethod)
    WriteResults(errorStatus)
    WriteResults(results.Details)
    If results.Origin.Function <> strMethod Then
      WriteResults(results.Origin.Class + "." + results.Origin.Function)
    End If
    If Not results.ResultException Is Nothing Then
      ' To view the stack trace on an exception uncomment the line below
      'WriteResults(results.ResultException.StackTrace)
    End If
    Environment.Exit(1)
  End Sub
  
  ' Write output data
  Sub WriteResults(content As String)
    ' Choose where to write out results
  
    ' Debug output
    'System.Diagnostics.Debug.WriteLine("ActivePDF: * " + content)
  
    ' Console
    Console.WriteLine(content)
  
    ' Log file
    'Using tw = New System.IO.StreamWriter(AppDomain.CurrentDomain.BaseDirectory & "application.log", True)
    '   tw.WriteLine("[" + DateTime.Now.ToString("MM/dd/yyyy HH:mm:ss") + "]: => " + content)
    'End Using
  End Sub
End Class